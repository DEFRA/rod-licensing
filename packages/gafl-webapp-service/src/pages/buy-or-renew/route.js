import { BUY_OR_RENEW } from '../../uri.js'
import Joi from 'joi'
import pageRoute from '../../routes/page-route.js'
import { nextPage } from '../../routes/next-page.js'
export const buyOrRenew = {
  buy: 'buy',
  renew: 'renew'
}

export const validator = Joi.object({
  'buy-or-renew': Joi.string()
    .valid('buy-licence', 'renew-licence')
    .required()
}).options({ abortEarly: false, allowUnknown: true })

const route = pageRoute(BUY_OR_RENEW.page, BUY_OR_RENEW.uri, validator, nextPage)

export default route
