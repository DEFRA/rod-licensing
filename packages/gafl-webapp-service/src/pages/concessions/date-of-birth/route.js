import { DATE_OF_BIRTH } from '../../../uri.js'
import Joi from 'joi'
import pageRoute from '../../../routes/page-route.js'
import { validation } from '@defra-fish/business-rules-lib'
import { nextPage } from '../../../routes/next-page.js'

export const validator = payload => {
  const dateOfBirth = `${payload['date-of-birth-year']}-${payload['date-of-birth-month']}-${payload['date-of-birth-day']}`
  Joi.assert(
    { 'date-of-birth': dateOfBirth },
    Joi.object({
      'date-of-birth': validation.contact.createBirthDateValidator(Joi)
    })
  )
}

export const getData = async request => {
  const { isLicenceForYou } = await request.cache().helpers.status.getCurrentPermission()

  return { isLicenceForYou }
}

export default pageRoute(DATE_OF_BIRTH.page, DATE_OF_BIRTH.uri, validator, nextPage, getData)
