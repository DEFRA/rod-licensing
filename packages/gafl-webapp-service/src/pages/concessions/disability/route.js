import { DISABILITY_CONCESSION } from '../../../uri.js'
import pageRoute from '../../../routes/page-route.js'
import Joi from 'joi'
import { validation } from '@defra-fish/business-rules-lib'
import * as concessionHelper from '../../../processors/concession-helper.js'
import { nextPage } from '../../../routes/next-page.js'
import { disabilityConcessionTypes } from './update-transaction.js'

const validator = Joi.object({
  'disability-concession': Joi.string()
    .valid(...Object.values(disabilityConcessionTypes))
    .required(),
  'ni-number': Joi.alternatives().conditional('disability-concession', {
    is: disabilityConcessionTypes.pipDla,
    then: validation.contact.createNationalInsuranceNumberValidator(Joi).required(),
    otherwise: Joi.string().empty('')
  }),
  'blue-badge-number': Joi.alternatives().conditional('disability-concession', {
    is: disabilityConcessionTypes.blueBadge,
    then: Joi.string()
      .max(25)
      .required(),
    otherwise: Joi.string().empty('')
  })
}).options({ abortEarly: false, allowUnknown: true })

export const getData = async request => {
  const permission = await request.cache().helpers.transaction.getCurrentPermission()

  return {
    hasJunior: concessionHelper.hasJunior(permission),
    hasSenior: concessionHelper.hasSenior(permission),
    isLicenceForYou: permission.isLicenceForYou,
    ...disabilityConcessionTypes
  }
}

export default pageRoute(DISABILITY_CONCESSION.page, DISABILITY_CONCESSION.uri, validator, nextPage, getData)
