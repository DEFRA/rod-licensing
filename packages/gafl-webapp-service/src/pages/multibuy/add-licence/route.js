import pageRoute from '../../../routes/page-route.js'
import Joi from 'joi'
import { ADD_LICENCE, CONTACT_SUMMARY, LICENCE_SUMMARY } from '../../../uri.js'
import { nextPage } from '../../../routes/next-page.js'

import GetDataRedirect from '../../../handlers/get-data-redirect.js'

export const getData = async request => {
  const status = await request.cache().helpers.status.getCurrentPermission()

  if (!status[LICENCE_SUMMARY.page]) {
    throw new GetDataRedirect(LICENCE_SUMMARY.uri)
  }

  if (!status[CONTACT_SUMMARY.page]) {
    throw new GetDataRedirect(CONTACT_SUMMARY.uri)
  }

  const transaction = await request.cache().helpers.transaction.get()

  return { numberOfLicences: transaction.permissions.length }
}

const validator = Joi.object({
  'add-licence': Joi.string()
    .valid('yes', 'no')
    .required()
}).options({ abortEarly: false, allowUnknown: true })

export default pageRoute(ADD_LICENCE.page, ADD_LICENCE.uri, validator, nextPage, getData)
