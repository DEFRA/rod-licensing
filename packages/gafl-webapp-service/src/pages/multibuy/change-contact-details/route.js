import pageRoute from '../../../routes/page-route.js'
import GetDataRedirect from '../../../handlers/get-data-redirect.js'
import { countries } from '../../../processors/refdata-helper.js'
import { HOW_CONTACTED } from '../../../processors/mapping-constants.js'
import { CHANGE_CONTACT_DETAILS_SEEN } from '../../../constants.js'
import { isPhysical } from '../../../processors/licence-type-display.js'
import { nextPage } from '../../../routes/next-page.js'
import { addLanguageCodeToUri } from '../../../processors/uri-helper.js'

import {
  CHANGE_CONTACT_DETAILS,
  ADDRESS_ENTRY,
  ADDRESS_SELECT,
  ADDRESS_LOOKUP,
  CONTACT,
  NEWSLETTER,
  LICENCE_FULFILMENT,
  LICENCE_CONFIRMATION_METHOD,
  CHANGE_LICENCE_OPTIONS
} from '../../../uri.js'

const CONTACT_TEXT_DEFAULT = {
  EMAIL: 'contact_summary_email',
  TEXT: 'contact_summary_text_sngl',
  DEFAULT: 'contact_summary_license_default'
}

const CONTACT_TEXT_NON_PHYSICAL = {
  EMAIL: 'contact_summary_email',
  TEXT: 'contact_summary_text_plrl',
  DEFAULT: 'contact_summary_license_non_physical'
}

const CONTACT_TEXT_PHYSICAL = {
  EMAIL: 'contact_summary_email',
  TEXT: 'contact_summary_text_plrl',
  DEFAULT: 'contact_summary_license_physical'
}

const CHANGE_CONTACT = 'change-contact'

class RowGenerator {
  constructor (request, permission) {
    this.request = request
    this.permission = permission
    this.labels = request.i18n.getCatalog()
  }

  _getPreferredMethodOfReminderText (contactTextSpec) {
    switch (this.permission.licensee.preferredMethodOfReminder) {
      case HOW_CONTACTED.email:
        return `${this.labels[contactTextSpec.EMAIL]}${this.permission.licensee.email}`
      case HOW_CONTACTED.text:
        return `${this.labels[contactTextSpec.TEXT]}${this.permission.licensee.mobilePhone}`
      default:
        return this.labels[contactTextSpec.DEFAULT]
    }
  }

  _getPreferredMethodOfConfirmation (contactTextSpec) {
    switch (this.permission.licensee.preferredMethodOfConfirmation) {
      case HOW_CONTACTED.email:
        return `${this.labels[contactTextSpec.EMAIL]}${this.permission.licensee.email}`
      case HOW_CONTACTED.text:
        return `${this.labels[contactTextSpec.TEXT]}${this.permission.licensee.mobilePhone}`
      default:
        return this.labels[contactTextSpec.DEFAULT]
    }
  }

  _generateRow (label, text, rawHref, visuallyHiddenText, id) {
    const href = addLanguageCodeToUri(this.request, rawHref)
    return {
      key: {
        text: label
      },
      value: {
        text
      },
      ...(href && {
        actions: {
          items: [
            {
              href,
              visuallyHiddenText,
              text: this.labels.contact_summary_change,
              attributes: { id }
            }
          ]
        }
      })
    }
  }

  generateStandardRow (label, text, rawHref, visuallyHiddenText, id) {
    return this._generateRow(this.labels[label], this.labels[text], rawHref, this.labels[visuallyHiddenText], id)
  }

  generateAddressRow (countryName) {
    const { licensee } = this.permission
    const text = [licensee.premises, licensee.street, licensee.locality, licensee.town, licensee.postcode, countryName?.toUpperCase()]
      .filter(Boolean)
      .join(', ')

    return this._generateRow(
      this.labels.contact_summary_row_address,
      text,
      ADDRESS_LOOKUP.uri,
      this.labels.contact_summary_hidden_address,
      'change-address'
    )
  }

  generateContactRow (label, href, visuallyHiddenText, id, contactTextSpec = CONTACT_TEXT_DEFAULT) {
    const contactText =
      label === 'contact_summary_row_contact'
        ? this._getPreferredMethodOfReminderText(contactTextSpec)
        : this._getPreferredMethodOfConfirmation(contactTextSpec)
    return this._generateRow(this.labels[label], contactText, href, this.labels[visuallyHiddenText], id)
  }
}

const checkNavigation = (status, permission) => {
  if (!permission.isRenewal) {
    if (!status[ADDRESS_ENTRY.page] && !status[ADDRESS_SELECT.page]) {
      throw new GetDataRedirect(ADDRESS_LOOKUP.uri)
    }

    if (!status[CONTACT.page]) {
      throw new GetDataRedirect(CONTACT.uri)
    }
  }

  if (isPhysical(permission)) {
    if (!status[LICENCE_FULFILMENT.page]) {
      throw new GetDataRedirect(LICENCE_FULFILMENT.uri)
    }
    if (!status[LICENCE_CONFIRMATION_METHOD.page]) {
      throw new GetDataRedirect(LICENCE_CONFIRMATION_METHOD.uri)
    }
  }
}

const getLicenseeDetailsSummaryRows = (permission, countryName, request) => {
  const rowGenerator = new RowGenerator(request, permission)

  const licenseeSummaryArray = [rowGenerator.generateAddressRow(countryName)]
  if (isPhysical(permission)) {
    if (permission.licensee.postalFulfilment) {
      licenseeSummaryArray.push(
        rowGenerator.generateStandardRow(
          'contact_summary_row_licence',
          'contact_summary_license_physical',
          LICENCE_FULFILMENT.uri,
          'contact_summary_hidden_licence_fulfilment',
          'change-licence-fulfilment-option'
        ),
        rowGenerator.generateContactRow(
          'contact_summary_row_licence_conf',
          LICENCE_CONFIRMATION_METHOD.uri,
          'contact_summary_hidden_licence_confirmation',
          'change-licence-confirmation-option'
        )
      )
    } else {
      licenseeSummaryArray.push(
        rowGenerator.generateContactRow(
          'contact_summary_row_licence',
          LICENCE_FULFILMENT.uri,
          'contact_summary_hidden_licence_confirmation',
          'change-licence-confirmation-option'
        )
      )
    }

    licenseeSummaryArray.push(
      rowGenerator.generateContactRow(
        'contact_summary_row_contact',
        CONTACT.uri,
        'contact_summary_hidden_contact',
        CHANGE_CONTACT,
        CONTACT_TEXT_PHYSICAL
      )
    )
  } else {
    licenseeSummaryArray.push(
      rowGenerator.generateContactRow(
        'contact_summary_row_licence_details',
        CONTACT.uri,
        'contact_summary_hidden_contact',
        CHANGE_CONTACT,
        CONTACT_TEXT_NON_PHYSICAL
      )
    )
  }

  if (permission.isLicenceForYou) {
    const text = permission.licensee.preferredMethodOfNewsletter === HOW_CONTACTED.none ? 'no' : 'yes'
    licenseeSummaryArray.push(
      rowGenerator.generateStandardRow(
        'contact_summary_row_newsletter',
        text,
        NEWSLETTER.uri,
        'contact_summary_hidden_newsletter',
        'change-newsletter'
      )
    )
  }

  return licenseeSummaryArray
}

const getData = async request => {
  const status = await request.cache().helpers.status.getCurrentPermission()
  const permission = await request.cache().helpers.transaction.getCurrentPermission()
  const mssgs = request.i18n.getCatalog()

  checkNavigation(status, permission)

  status.changeContactDetails = CHANGE_CONTACT_DETAILS_SEEN.SEEN

  console.log(status.changeContactDetails)

  await request.cache().helpers.status.setCurrentPermission(status)
  const countryName = await countries.nameFromCode(permission.licensee.countryCode)

  const changeLicenceDetails = permission.isLicenceForYou ? mssgs.change_licence_details_you : mssgs.change_licence_details_other

  return {
    summaryTable: getLicenseeDetailsSummaryRows(permission, countryName, request),
    uri: {
      changeLicenceOptions: CHANGE_LICENCE_OPTIONS.uri
    },
    changeLicenceDetails
  }
}

export default pageRoute(CHANGE_CONTACT_DETAILS.page, CHANGE_CONTACT_DETAILS.uri, null, nextPage, getData)
