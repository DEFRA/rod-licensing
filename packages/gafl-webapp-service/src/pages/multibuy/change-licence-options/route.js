import pageRoute from '../../../routes/page-route.js'
import GetDataRedirect from '../../../handlers/get-data-redirect.js'
import { licenceTypeDisplay, getErrorPage } from '../../../processors/licence-type-display.js'
import { displayStartTime, cacheDateFormat } from '../../../processors/date-and-time-display.js'
import findPermit from '../../summary/find-permit.js'
import { CHANGE_LICENCE_OPTIONS_SEEN } from '../../../constants.js'
import moment from 'moment-timezone'
import {
  NAME,
  CHANGE_LICENCE_OPTIONS,
  CHANGE_CONTACT_DETAILS,
  LICENCE_LENGTH,
  LICENCE_TYPE,
  LICENCE_TO_START,
  DISABILITY_CONCESSION,
  DATE_OF_BIRTH,
  RENEWAL_START_DATE,
  NEW_TRANSACTION
} from '../../../uri.js'
import { nextPage } from '../../../routes/next-page.js'
import { START_AFTER_PAYMENT_MINUTES, SERVICE_LOCAL_TIME } from '@defra-fish/business-rules-lib'
import { CONCESSION, CONCESSION_PROOF } from '../../../processors/mapping-constants.js'
import * as concessionHelper from '../../../processors/concession-helper.js'

export const getData = async request => {
  const status = await request.cache().helpers.status.getCurrentPermission()
  const permission = await request.cache().helpers.transaction.getCurrentPermission()

  if (!status.renewal) {
    /*
     * Before we try and filter the permit it is necessary to check that the user has navigated through
     * the journey in such a way as to have gather all the required data. They may have manipulated the
     * journey by typing into the address bar in which case they will be redirected back to the
     * appropriate point in the journey. For a renewal this is not necessary.
     */
    const errorOnPage = getErrorPage(permission)
    if (errorOnPage) {
      throw new GetDataRedirect(errorOnPage)
    }
  }

  status.fromLicenceOptions = CHANGE_LICENCE_OPTIONS_SEEN.SEEN
  await request.cache().helpers.status.setCurrentPermission(status)
  await findPermit(permission, request)
  const startTimeString = displayStartTime(request, permission)

  return {
    permission,
    startTimeString,
    startAfterPaymentMinutes: START_AFTER_PAYMENT_MINUTES,
    licenceTypeStr: licenceTypeDisplay(permission, request.i18n.getCatalog()),
    isRenewal: status.renewal,
    isContinuing: !!(permission.renewedEndDate && permission.renewedEndDate === permission.licenceStartDate),
    hasExpired: moment(moment().tz(SERVICE_LOCAL_TIME)).isAfter(moment(permission.renewedEndDate, cacheDateFormat)),
    disabled: permission.concessions && permission.concessions.find(c => c.type === CONCESSION.DISABLED),
    concessionProofs: CONCESSION_PROOF,
    hasJunior: concessionHelper.hasJunior(permission),
    cost: permission.permit.cost,
    birthDateStr: moment(permission.licensee.birthDate, cacheDateFormat).format('Do MMMM YYYY'),
    uri: {
      name: NAME.uri,
      licenceLength: LICENCE_LENGTH.uri,
      licenceType: LICENCE_TYPE.uri,
      licenceToStart: LICENCE_TO_START.uri,
      dateOfBirth: DATE_OF_BIRTH.uri,
      disabilityConcession: DISABILITY_CONCESSION.uri,
      licenceStartDate: status.renewal ? RENEWAL_START_DATE.uri : LICENCE_TO_START.uri,
      clear: NEW_TRANSACTION.uri,
      changeContactDetails: CHANGE_CONTACT_DETAILS.uri
    }
  }
}

export default pageRoute(CHANGE_LICENCE_OPTIONS.page, CHANGE_LICENCE_OPTIONS.uri, null, nextPage, getData)
