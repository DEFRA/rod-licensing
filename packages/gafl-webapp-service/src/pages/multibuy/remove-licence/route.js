import pageRoute from '../../../routes/page-route.js'
import Joi from 'joi'
import { REMOVE_LICENCE, VIEW_LICENCES } from '../../../uri.js'
import { nextPage } from '../../../routes/next-page.js'
import { licenceTypeDisplay, licenceTypeAndLengthDisplay } from '../../../processors/licence-type-display.js'
import { displayStartTime } from '../../../processors/date-and-time-display.js'
import { addLanguageCodeToUri } from '../../../processors/uri-helper.js'

export const getData = async request => {
  await request.cache().helpers.status.set({ currentPermissionIdx: parseInt(request.query.permissionIndex) })
  const permission = await request.cache().helpers.transaction.getCurrentPermission()
  const mssgs = request.i18n.getCatalog()

  return {
    licenceHolder: `${permission.licensee.firstName} ${permission.licensee.lastName}`,
    type: licenceTypeDisplay(permission, mssgs),
    length: licenceTypeAndLengthDisplay(permission, mssgs),
    start: displayStartTime(request, permission),
    price: permission.permit.cost,
    uri: {
      view_licences: addLanguageCodeToUri(request, VIEW_LICENCES.uri)
    }
  }
}

export const validator = Joi.object({}).options({ abortEarly: false, allowUnknown: true })

export default pageRoute(REMOVE_LICENCE.page, REMOVE_LICENCE.uri, validator, nextPage, getData)
