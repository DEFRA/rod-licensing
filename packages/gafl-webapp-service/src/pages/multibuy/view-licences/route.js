import pageRoute from '../../../routes/page-route.js'
import Joi from 'joi'
import { VIEW_LICENCES } from '../../../uri.js'
import { licenceTypeDisplay, licenceTypeAndLengthDisplay } from '../../../processors/licence-type-display.js'
import { displayStartTime } from '../../../processors/date-and-time-display.js'
import { nextPage } from '../../../routes/next-page.js'
import { checkDuplicates } from '../../../handlers/multibuy-duplicate-handler.js'

export const getData = async request => {
  const transaction = await request.cache().helpers.transaction.get()
  const mssgs = request.i18n.getCatalog()

  const licences = transaction.permissions.map((permission, index) => ({
    licenceHolder: `${permission.licensee.firstName} ${permission.licensee.lastName}`,
    type: licenceTypeDisplay(permission, mssgs),
    length: licenceTypeAndLengthDisplay(permission, mssgs),
    start: displayStartTime(request, permission),
    price: permission.permit.cost,
    index
  }))

  const duplicate = await checkDuplicates(licences)

  return { licences, duplicate }
}

export const validator = Joi.object({
  agree: Joi.string().valid('yes').required()
}).options({ abortEarly: false, allowUnknown: true })

export default pageRoute(VIEW_LICENCES.page, VIEW_LICENCES.uri, validator, nextPage, getData)
