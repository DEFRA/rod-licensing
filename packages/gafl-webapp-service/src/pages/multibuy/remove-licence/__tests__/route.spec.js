import { getData, validator } from '../route.js'
import { createMockRequest } from '../../../../__mocks__/request.js'
import pageRoute from '../../../../routes/page-route.js'
import { nextPage } from '../../../../routes/next-page.js'
import { VIEW_LICENCES } from '../../../../uri.js'
import { licenceTypeDisplay, licenceTypeAndLengthDisplay } from '../../../../processors/licence-type-display.js'
import { displayStartTime } from '../../../../processors/date-and-time-display.js'
import { addLanguageCodeToUri } from '../../../../processors/uri-helper.js'

jest.mock('../../../../processors/uri-helper.js', () => ({
  addLanguageCodeToUri: jest.fn((_request, uri) => uri)
}))
jest.mock('../../../../processors/licence-type-display.js')
jest.mock('../../../../processors/date-and-time-display.js')
jest.mock('../../../../routes/page-route.js')

const permission = {
  licensee: {
    firstName: 'Turanga',
    lastName: 'Leela'
  },
  licenceType: 'trout-and-coarse',
  numberOfRods: '2',
  licenceLength: '8D',
  permit: { cost: 12 }
}

const catalog = Symbol('mock catalog')

describe('remove-licence > default', () => {
  it('should call the pageRoute with remove-licence, /buy/remove-licence, validator and nextPage', async () => {
    expect(pageRoute).toBeCalledWith('remove-licence', '/buy/remove-licence', validator, nextPage, getData)
  })
})

describe('remove-licence > getData', () => {
  const mockRequest = createMockRequest({
    cache: {
      transaction: {
        permissions: [permission]
      }
    }
  })

  const getSampleRequest = () => ({
    ...mockRequest,
    i18n: {
      getCatalog: () => catalog
    }
  })

  licenceTypeDisplay.mockReturnValue('Trout and coarse, up to 2 rods')
  licenceTypeAndLengthDisplay.mockReturnValue('8 days')
  displayStartTime.mockReturnValue('9:32am on 23 June 2021')
  addLanguageCodeToUri.mockReturnValue('/buy/remove-licence')

  it('expected data from getData', async () => {
    const data = await getData(getSampleRequest())
    expect(data).toMatchSnapshot()
  })
})

describe('getData', () => {
  const mockRequest = createMockRequest({
    cache: {
      transaction: {
        permissions: [permission]
      }
    }
  })

  const getSampleRequest = () => ({
    ...mockRequest,
    i18n: {
      getCatalog: () => catalog
    }
  })

  beforeEach(() => jest.clearAllMocks())

  it('licenceTypeDisplay is called with the expected arguments', async () => {
    await getData(getSampleRequest())

    expect(licenceTypeDisplay).toHaveBeenCalledWith(permission, catalog)
  })

  it('return value of licenceTypeDisplay is used for type', async () => {
    const returnValue = Symbol('return value')
    licenceTypeDisplay.mockReturnValueOnce(returnValue)

    const result = await getData(getSampleRequest())

    expect(result.type).toEqual(returnValue)
  })

  it('licenceTypeAndLengthDisplay is called with the expected arguments', async () => {
    await getData(getSampleRequest())
    expect(licenceTypeAndLengthDisplay).toHaveBeenCalledWith(permission, catalog)
  })

  it('return value of licenceTypeAndLengthDisplay is used for length', async () => {
    const returnValue = Symbol('return value')
    licenceTypeAndLengthDisplay.mockReturnValueOnce(returnValue)

    const result = await getData(getSampleRequest())

    expect(result.length).toEqual(returnValue)
  })

  it('displayStartTime is called with the expected arguments', async () => {
    const sampleRequest = getSampleRequest()
    await getData(sampleRequest)
    expect(displayStartTime).toHaveBeenCalledWith(sampleRequest, permission)
  })

  it('return value of displayStartTime is used for start', async () => {
    const returnValue = Symbol('return value')
    displayStartTime.mockReturnValueOnce(returnValue)

    const result = await getData(getSampleRequest())

    expect(result.start).toEqual(returnValue)
  })

  it('addLanguageCodeToUri is called with the expected arguments', async () => {
    const sampleRequest = getSampleRequest()
    await getData(sampleRequest)
    expect(addLanguageCodeToUri).toHaveBeenCalledWith(sampleRequest, VIEW_LICENCES.uri)
  })

  it('return value of addLanguageCodeToUri is used for uri.view_licences', async () => {
    const returnValue = Symbol('return value')
    addLanguageCodeToUri.mockReturnValueOnce(returnValue)

    const result = await getData(getSampleRequest())

    expect(result.uri.view_licences).toEqual(returnValue)
  })
})
