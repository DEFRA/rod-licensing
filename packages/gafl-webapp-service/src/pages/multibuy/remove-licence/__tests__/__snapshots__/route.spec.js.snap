// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remove-licence > getData expected data from getData 1`] = `
Object {
  "length": "8 days",
  "licenceHolder": "Turanga Leela",
  "price": 12,
  "start": "9:32am on 23 June 2021",
  "type": "Trout and coarse, up to 2 rods",
  "uri": Object {
    "view_licences": "/buy/remove-licence",
  },
}
`;
